traffic_ops/traffic_ops_golang/perlconfig.go

```diff
@@ -0,0 +1,288 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/apache/incubator-trafficcontrol/traffic_monitor_golang/common/log"
+)
+
+const OldAccessLogPath = "/var/log/traffic_ops/access.log"
+const NewLogPath = "/var/log/traffic_ops/traffic_ops_golang.log"
+
+func GetPerlConfigs(cdnConfPath string, dbConfPath string) (Config, error) {
+	configBytes, err := ioutil.ReadFile(cdnConfPath)
+	if err != nil {
+		return Config{}, fmt.Errorf("reading CDN conf '%v': %v", cdnConfPath, err)
+	}
+	dbConfBytes, err := ioutil.ReadFile(dbConfPath)
+	if err != nil {
+		return Config{}, fmt.Errorf("reading db conf '%v': %v", dbConfPath, err)
+	}
+	return getPerlConfigsFromStrs(string(configBytes), string(dbConfBytes))
+}
+
+func getPerlConfigsFromStrs(cdnConfBytes string, dbConfBytes string) (Config, error) {
+	cfg, err := getCDNConf(cdnConfBytes)
+	if err != nil {
+		return Config{}, fmt.Errorf("parsing CDN conf '%v': %v", cdnConfBytes, err)
+	}
+
+	dbconf, err := getDbConf(string(dbConfBytes))
+	if err != nil {
+		return Config{}, fmt.Errorf("parsing db conf '%v': %v", dbConfBytes, err)
+	}
+	cfg.DBUser = dbconf.User
+	cfg.DBPass = dbconf.Password
+	cfg.DBServer = dbconf.Hostname
+	cfg.DBDB = dbconf.DBName
+	cfg.DBSSL = false // TODO fix
+	if dbconf.Port != "" {
+		cfg.DBServer += ":" + dbconf.Port
+	}
+
+	cfg.LogLocationInfo = OldAccessLogPath
+	cfg.LogLocationError = NewLogPath
+	cfg.LogLocationWarning = NewLogPath
+	cfg.LogLocationEvent = NewLogPath
+	cfg.LogLocationDebug = log.LogLocationNull
+
+	return cfg, nil
+}
+
+func getCDNConf(s string) (Config, error) {
+	cfg := Config{}
+	obj, err := ParsePerlObj(s)
+	if err != nil {
+		return Config{}, fmt.Errorf("parsing Perl object: %v", err)
+	}
+
+	if cfg.HTTPPort, err = getPort(obj); err != nil {
+		return Config{}, err
+	}
+
+	if cfg.TOSecret, err = getSecret(obj); err != nil {
+		return Config{}, err
+	}
+
+	oldPort, err := getOldPort(obj)
+	if err != nil {
+		return Config{}, err
+	}
+	cfg.TOURLStr = "https://127.0.0.1:" + oldPort
+	if cfg.TOURL, err = url.Parse(cfg.TOURLStr); err != nil {
+		return Config{}, fmt.Errorf("Invalid Traffic Ops URL '%v': err", cfg.TOURL, err)
+	}
+
+	cfg.CertPath, err = getConfigCert(obj)
+	if err != nil {
+		return Config{}, err
+	}
+
+	cfg.KeyPath, err = getConfigKey(obj)
+	if err != nil {
+		return Config{}, err
+	}
+
+	return cfg, nil
+}
+
+func getPort(obj map[string]interface{}) (string, error) {
+	portStrI, ok := obj["traffic_ops_golang_port"]
+	if !ok {
+		return "", fmt.Errorf("missing traffic_ops_golang_port key")
+	}
+	portStr, ok := portStrI.(string)
+	if !ok {
```

Comment:
It's a string just because it was easier to work with in the code, e.g. the Go HTTP server takes a string. I can make it an `int` or `uint` if you want. I'd rather not `uint16` though, it's unusual, and even if performance mattered it's not any faster on a 64-bit processor.